### PR TITLE
Improvement/auth session logic

### PR DESCRIPTION
**Pull request by ruig1992**
[trs Store auth token in the global state](https://trello.com/c/5TaxtW1r)
[trs Hide functionality depending on the user role](https://trello.com/c/8wCSHVDb)
[trs Improved the auth service logic](https://trello.com/c/Y5UCS3wy)

### CODE REVIEW

**Comment by ruig1992** [comment]
This awful ugly code is now changing in the hope of a better use.

**Review by sergey-ben — DISMISSED** [review]

**Comment by ruig1992** [comment]
This code is much better.)

File: resources/assets/js/app/services/AuthService.js [hunk 1]
```diff
@@ -1,77 +1,110 @@
...
+            store = params.store;
+            loginSuccess = params.loginSuccess;
+        },
+
+        getFromState(param) {
+            return store.getState().user.session[param];
+        },
+
+        getSessionToken() {
+            return _this.getFromState('token') || storage[tokenKeyName];
```
Comment (TatsuUkraine) [inline resources/assets/js/app/services/AuthService.js]: you should get token always from global state, localStorage should be used just as initial source of data. on app init you need get token from localStorage and put it in global state, on destroy session you need remove it from there and from global state, on initSession - put it in localStorage and global state. you can dispatch actions from this service to update state

Comment (TatsuUkraine) [reply resources/assets/js/app/services/AuthService.js]: in this case localStorage should be source of data only for page reload, it shouldn't be main source of data

**Review by TatsuUkraine — CHANGES_REQUESTED** [review]

**Review by TatsuUkraine — DISMISSED** [review]

**Review by TatsuUkraine — APPROVED** [review]

**Pull request merged**